### PR TITLE
feat(patch): [sc-3762] Make Dictionary as Transferable.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

Make Dictionary as Transferable.

## How Has This Been Tested?

Unit tests.

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/DistributedSystemConformance/Dictionary+Transferable.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import Helpers
+
+fileprivate typealias SizeType = UInt32
+
+public extension Dictionary where Key: Serializable, Value: Serializable {
+    func withUnsafeBytesSerialization<Result>(_ body: (UnsafeRawBufferPointer) throws -> Result) rethrows -> Result {
+        var buffer = UnsafeMutableRawBufferPointer(start: nil, count: 0)
+        defer {
+            if buffer.baseAddress != nil {
```
Comment (dimlio) [inline Sources/DistributedSystemConformance/Dictionary+Transferable.swift:17]: I believe this check is not necessary.

File: Sources/DistributedSystemConformance/Dictionary+Transferable.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                buffer.deallocate()
+            }
+        }
+        var pos = 0
+        for entry in self {
```
Comment (dimlio) [inline Sources/DistributedSystemConformance/Dictionary+Transferable.swift:22]:
```suggestion
        for (key, value) in self {
```
?

File: Tests/DistributedSystemTests/DistributedSystemTests.swift [hunk 3]
```diff
@@ -49,6 +58,25 @@ class Service: TestableService {
...
         fatalError("Should not be called")
     }
+
+    func handleMonsters(_ monsters: [Monster]) async {
+        logger.info("SERVICE: got \(monsters.count) monsters")
+    }
+
+    func handleMonsters(_ monsters: [String: Monster]) async {
+        var invalidEntries = 0
+        for entry in monsters {
```
Comment (dimlio) [inline Tests/DistributedSystemTests/DistributedSystemTests.swift:68]:
```suggestion
        for key in monsters.keys {
```

**Review by dimlio — APPROVED** [review]

File: Tests/DistributedSystemTests/DistributedSystemTests.swift (hunk 3 above)
Comment (ser-0xff) [reply Tests/DistributedSystemTests/DistributedSystemTests.swift:68]: Fixed.

File: Sources/DistributedSystemConformance/Dictionary+Transferable.swift (hunk 1 above)
Comment (ser-0xff) [reply Sources/DistributedSystemConformance/Dictionary+Transferable.swift:17]: You are right, fixed.

**Pull request merged**